src/main/java/com/ays/admin_user/service/impl/AdminUserServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.ays.admin_user.service.impl;
+
+import com.ays.admin_user.model.AdminUser;
+import com.ays.admin_user.model.dto.request.AdminUserListRequest;
+import com.ays.admin_user.model.entity.AdminUserEntity;
+import com.ays.admin_user.model.mapper.AdminUserEntityToAdminUserMapper;
+import com.ays.admin_user.repository.AdminUserRepository;
+import com.ays.admin_user.service.AdminUserService;
+import com.ays.auth.model.AysIdentity;
+import com.ays.auth.model.enums.AysUserType;
+import com.ays.common.model.AysPage;
+import com.ays.common.model.AysSpecification;
+import com.ays.common.util.exception.AysUnexpectedArgumentException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This service class implements the {@link AdminUserService} interface and provides methods for
+ * performing admin operations. It uses the {@link AdminUserRepository} and the
+ * {@link AdminUserEntityToAdminUserMapper} for mapping the request to entity objects.
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AdminUserServiceImpl implements AdminUserService {
+
+    private final AdminUserRepository adminUserRepository;
+
+    private final AdminUserEntityToAdminUserMapper adminEntityToAdminMapper = AdminUserEntityToAdminUserMapper.initialize();
+
+    private final AysIdentity identity;
+
+    /**
+     * Get All admin users based on the given {@link AdminUserListRequest} object. First, it is determined whether
```

Comment:
We should correct the explanation here as we are not retrieving all users, so we shouldn't state "it retrieves all of them."

Reply:
Ok. I revised the explanation. Can you look through it again?